life_expectancy/main.py

```diff
@@ -1,33 +1,45 @@
 """This is the main module"""
 from pathlib import Path
-
 import argparse
 import pandas as pd
-
-from life_expectancy.cleaning import clean_data
-from life_expectancy.load_save_data import load_data, save_data
-
+from life_expectancy.cleaning import filter_data, clean_data, feature_cleaning
+from life_expectancy.load_save_data import TSVFileReader, JSONFileReader, FileProcessor, save_data
+from life_expectancy.countries import Region
 
 script_dir = Path(__file__).resolve().parent
-input_file_path = script_dir/"data"/"eu_life_expectancy_raw.tsv"
+input_file_path_tsv= script_dir/"data"/"eu_life_expectancy_raw.tsv"
+input_file_path_json = script_dir/"data"/"eurostat_life_expect.zip"
 output_file_path = script_dir/"data"/"pt_life_expectancy.csv"
 
 
-def main(region: str = 'PT') -> pd.DataFrame:
+def main(region: Region = Region.PT) -> pd.DataFrame:
     """
     Main function that load, cleans and save a cleaned version of the life expectancy data
     Args:
-        region (str): String with region information
+        region (Regions): Enum with region information
     """
-    life_exp_raw_data = load_data(input_file_path)
-    life_exp = clean_data(life_exp_raw_data, region)
+    # Verify if the region is in the list of Contries and Regions
+    if region.value not in Region.__members__:
+        raise ValueError("This is an Invalid Country")
+
+    file_reader = JSONFileReader()
+    file_processor = FileProcessor(file_reader)
+    life_exp_raw_data = file_processor.processor_file(input_file_path_json)
+
+    if isinstance(file_reader, TSVFileReader):
+        life_exp = clean_data(life_exp_raw_data)
+    else:
+        life_exp = feature_cleaning(life_exp_raw_data)
+
+    life_exp = filter_data(life_exp, region)
+
     save_data(life_exp, output_file_path)
 
     return life_exp
 
 
 if __name__ == '__main__': # pragma: no cover
     parser = argparse.ArgumentParser(description="Process all the arguments for this cleaning")
-    parser.add_argument("--region", default="PT", help="Filters data for this region", type=str)
+    parser.add_argument("--region", default=Region.PT, help="Filters data for this region")
```

Comment:
This is incomplete: args.region will return a str not a Region.

But if you use it like this, then not only you'll get a region, but you will have the Enum's built-in validation checking if the input was valid:

```python
from enum import Enum
from argparse import ArgumentParser

class Color(Enum):
    red = 'red'
    blue = 'blue'

parser = ArgumentParser()
parser.add_argument('color', type=Color)

# Testing:
args = parser.parse_args(["red"])
print('Your color was:', args.color)
# Will return Your color was: Color.red

args = parser.parse_args(["black"])
# Will raise error: argument color: invalid Color value: 'black'
```

Reply:
Thanks, added type = Region :)